codemods/icons.js

```diff
@@ -0,0 +1,86 @@
+const visit = require('unist-util-visit');
+const {
+  findAttribute,
+  hasClassName,
+  isMdxElement,
+  removeAttribute,
+  addAttribute,
+} = require('./utils/mdxast');
+
+const ICON_NAME = /\bfa-((?!\d)[a-z-]+\b)/;
+const ICON_SIZE = /fa-(\d)x/;
+
+const FA_TO_FEATHER_ICONS = {
+  'arrows-h': 'move-horizontal',
+  'bar-chart': 'bar-chart-2',
+  bolt: 'zap',
+  'caret-down': 'chevron-down',
+  'check-square-o': 'check-square',
+  cog: 'settings',
+  'commenting-o': 'message-circle',
+  database: 'database',
+  exchange: 'repeat',
+  'exclamation-triangle': 'alert-triangle',
+  filter: 'filter',
+  'life-ring': 'life-buoy',
+  'list-ol': 'list',
+  'minus-circle': 'minus-circle',
+  sliders: 'sliders',
+  terminal: 'terminal',
+  wrench: 'tool',
+};
+
+const icons = () => (tree, file) => {
+  visit(
+    tree,
+    (node) => isMdxElement('i', node) && hasClassName('fa', node),
+    (icon) => {
+      const className = findAttribute('className', icon);
+      const [, size] = className.match(ICON_SIZE) || [];
```

Comment:
In the event that there's no match (`[]`) wouldn't destructuring the second item in the array return `undefined`? Perhaps something like this might be safer:

```jsx
const size = classname.match(ICON_SIZE)?.[1] || 0;
```

Reply:
Yes, `size`, would be `undefined` in this case. I am [handling that below](https://github.com/newrelic/docs-website/pull/123/files#diff-26d8a9a27464d33f0b45555b8697c54fR79-R81) by checking it to see if its a truthy value before adding it as an attribute:

```js
if (size) {
  // ...
}
```

Reply:
Also, unfortunately, because we are in node-land, we can't use the fancy optional chaining syntax 😞